tools/package_tools/mmdeploy_builder.py

```diff
@@ -0,0 +1,354 @@
+
+
+def clear_mmdeploy(mmdeploy_dir: str):
+    logging.info(f'cleaning mmdeploy: {mmdeploy_dir}')
+
+    def _remove_in_mmdeploy(path):
+        remove_dir = osp.join(mmdeploy_dir, path)
+        _remove_if_exist(remove_dir)
+
+    # remove build file
+    _remove_in_mmdeploy('build')
+
+    # remove dist
+    _remove_in_mmdeploy('dist')
+
+    # remove installed library
+    _remove_in_mmdeploy('mmdeploy/lib')
+
+    # remove onnx2ncnn and ncnn ext
+    _remove_in_mmdeploy('mmdeploy/backend/ncnn/onnx2ncnn')
+    _remove_in_mmdeploy('mmdeploy/backend/ncnn/onnx2ncnn.exe')
+    ncnn_ext_paths = glob(
+        osp.join(mmdeploy_dir, 'mmdeploy/backend/ncnn/ncnn_ext.*'))
+    for ncnn_ext_path in ncnn_ext_paths:
+        os.remove(ncnn_ext_path)
+
+
+def build_mmdeploy(cfg, mmdeploy_dir, dist_dir=None):
+    cmake_flags = cfg.get('cmake_flags', [])
+    cmake_envs = cfg.get('cmake_envs', dict())
+
+    args = [f'-D{k}={v}' for k, v in cmake_envs.items()]
+
+    # clear mmdeploy
+    clear_mmdeploy(mmdeploy_dir)
+
+    build_dir = osp.join(mmdeploy_dir, 'build')
+    if not osp.exists(build_dir):
+        os.mkdir(build_dir)
+
+        # cmake cmd
+        cmake_cmd = ' '.join(['cmake ..'] + cmake_flags + args)
+        _call_command(cmake_cmd, build_dir)
+
+    if sys.platform == 'win32':
+        # build cmd
+        build_cmd = 'cmake --build . --config Release -- /m'
+        _call_command(build_cmd, build_dir)
+        install_cmd = 'cmake --install . --config Release'
+        _call_command(install_cmd, build_dir)
+    else:
+        # build cmd
+        build_cmd = 'cmake --build . -- -j$(nproc) && cmake --install .'
+        _call_command(build_cmd, build_dir)
+
+    # build wheel
+    bdist_cmd = _create_bdist_cmd(cfg, c_ext=False, dist_dir=dist_dir)
+    _call_command(bdist_cmd, mmdeploy_dir)
+
+
+def get_dir_name(cfg, tag, default_name):
+    if tag not in cfg:
+        logging.warning(f'{tag} not found, use `{default_name}` as default.')
+    else:
+        default_name = cfg[tag]
+    return cfg, default_name
+
+
+def check_env(cfg: Dict):
+    env_info = {}
+
+    cmake_envs = cfg.get('cmake_envs', dict())
+
+    # system
+    platform_system = platform.system().lower()
+    platform_machine = platform.machine().lower()
+    env_info['system'] = platform_system
+    env_info['machine'] = platform_machine
+
+    # CUDA version
+    cuda_version = 'unknown'
+
+    CUDA_TOOLKIT_ROOT_DIR = cmake_envs.get('CUDA_TOOLKIT_ROOT_DIR', '')
+    CUDA_TOOLKIT_ROOT_DIR = osp.expandvars(CUDA_TOOLKIT_ROOT_DIR)
+    nvcc_cmd = 'nvcc' if len(CUDA_TOOLKIT_ROOT_DIR) <= 0 else osp.join(
+        CUDA_TOOLKIT_ROOT_DIR, 'bin', 'nvcc')
+
+    if osp.exists(nvcc_cmd):
+        nvcc = check_output(f'"{nvcc_cmd}" -V', shell=True)
+        nvcc = nvcc.decode('utf-8').strip()
+        pattern = r'Cuda compilation tools, release (\d+.\d+)'
+        match = re.search(pattern, nvcc)
+        if match is not None:
+            cuda_version = match.group(1)
```

Comment:
On windows, there is no CUDA_TOOLKIT_ROOT_DIR variable in windows_x64.xml, so it will use nvcc_cmd = 'nvcc'. It doesn't matter because the installation of cuda will add the nvcc bin folder to the path. 

While check osp.exists(nvcc_cmd) will fail on windows for two reasons: one is the nvcc_cmd is not a absoluate path, the other is that the executable file should be nvcc.exe other than nvcc.  May use try except there.